airbyte-server/src/main/java/io/airbyte/server/handlers/WorkspacesHandler.java

```diff
@@ -24,37 +24,141 @@
+
+    final StandardWorkspace workspace = new StandardWorkspace()
+        .withWorkspaceId(UUID.randomUUID())
+        .withCustomerId(UUID.randomUUID())
+        .withName(workspaceCreate.getName())
+        .withSlug(slugify.slugify(workspaceCreate.getName()))
+        .withInitialSetupComplete(initialSetupComplete != null ? initialSetupComplete : false)
+        .withAnonymousDataCollection(anonymousDataCollection != null ? anonymousDataCollection : false)
+        .withNews(news != null ? news : false)
+        .withSecurityUpdates(securityUpdates != null ? securityUpdates : false)
+        .withDisplaySetupWizard(displaySetupWizard != null ? displaySetupWizard : false)
+        .withTombstone(false);
+
+    if (!Strings.isNullOrEmpty(email)) {
+      workspace.withEmail(email);
+    }
+
+    try {
+      if (getWorkspaceBySlug(workspace.getSlug()) != null) {
+        throw new KnownException(HttpStatus.CONFLICT_409, "A workspace already exists with the same name");
+      }
+    } catch (NotFoundException e) {
+      // if the workspace does not exist, we can create a new one
+    }
+
+    configRepository.writeStandardWorkspace(workspace);
+
+    return buildWorkspaceRead(workspace);
+  }
+
+  public void deleteWorkspace(final WorkspaceIdRequestBody workspaceIdRequestBody)
+      throws JsonValidationException, IOException, ConfigNotFoundException {
+    // get existing implementation
+    final StandardWorkspace persistedWorkspace = configRepository.getStandardWorkspace(workspaceIdRequestBody.getWorkspaceId());
+
+    // disable all connections associated with this workspace
+    for (ConnectionRead connectionRead : connectionsHandler.listConnectionsForWorkspace(workspaceIdRequestBody).getConnections()) {
+      connectionsHandler.deleteConnection(connectionRead);
+    }
+
+    // disable all destinations associated with this workspace
+    for (DestinationRead destinationRead : destinationHandler.listDestinationsForWorkspace(workspaceIdRequestBody).getDestinations()) {
+      destinationHandler.deleteDestination(destinationRead);
+    }
+
+    // disable all sources associated with this workspace
+    for (SourceRead sourceRead : sourceHandler.listSourcesForWorkspace(workspaceIdRequestBody).getSources()) {
+      sourceHandler.deleteSource(sourceRead);
+    }
+
+    persistedWorkspace.withTombstone(true);
+    configRepository.writeStandardWorkspace(persistedWorkspace);
   }
 
   public WorkspaceRead getWorkspace(WorkspaceIdRequestBody workspaceIdRequestBody)
       throws JsonValidationException, IOException, ConfigNotFoundException {
-    return buildWorkspaceReadFromId(workspaceIdRequestBody.getWorkspaceId());
+    final UUID workspaceId = workspaceIdRequestBody.getWorkspaceId();
+    final StandardWorkspace workspace = configRepository.getStandardWorkspace(workspaceId);
+
+    if (workspace.getTombstone()) {
+      throw new NotFoundException("Could not find workspace");
```

Comment:
Same here. We can add a boolean to `getStandardWorkspace` to indicate if you want to include the tombstone ones or not. This way you can let the config repo throw the `ConfigNotFoundException`.

WDYT?

Reply:
The good thing about adding that boolean is that people will have to make the conscious choice of including tombstoned or not. Less risk of forgetting to filter out the tombstoned